app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -211,6 +213,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
     }
 
     final override fun onBackPressed() {
+        FullscreenTransitionMeasurement.onBackPressed() // DO NOT MOVE ANYTHING ABOVE HERE.
```

Comment:
is there a reason why we can't move anything above here? Just wondering.

Reply:
If code is moved above here and it takes a long time, it won't be included in the measurement: we want the measurement to happen first.

I wonder if there's a more foolproof way to do this (e.g. with annotations?).

Reply:
So, I think with anything like this, we shouldn't couple performance code with our app code. I was thinking more of an AOP way ( I'm thinking the way Spring does things) but looking around quickly I found this article: https://medium.com/dont-code-me-on-that/aspect-oriented-programming-in-kotlin-95ff8598913 which highlights the way we can do it in kotlin :) THoughts?

Reply:
Oh interesting: that's pretty neat and avoids the footgun of someone moving code where they're not supposed to. Sure, let's aim for something like that in the future (though I'm going to do a little more reading on AOP to understand it more holistically).

Reply:
Yeah for sure, if we don't want to do this here, I think we can go ahead with this PR. It helps with less coupling. 😄

Reply:
We're not intending to land this: sorry if that wasn't clear. The code is just here for you to take measurements.